File: Chapter-5-Classes-and-Objects/010-a-sample-program-to-introduce-friend-function.cpp
// a sample program to introduce friend function
#include <bits/stdc++.h>

using namespace std;

class sample
{
    int a;
    int b;

public:
    void setValue()
    {
        a = 25;
        b = 40;
    }
    friend float mean(sample s);
};

float mean(sample s)
{
    return float(s.a + s.b) / 2.0;
}

int main()
{
    sample X; /// object X;
    X.setValue();
    cout << "Mean value : " << mean(X) << endl;
}

// output
// Mean value : 32.5

File: Chapter-5-Classes-and-Objects/010-a-sample-program-to-introduce-friend-function.py
class Sample:
    def __init__(self):
        self.a = 0
        self.b = 0

    def set_value(self, a, b):
        self.a = a
        self.b = b
    
def mean(s):
    return float(s.a + s.b) / 2.0

if __name__ == '__main__':
    X = Sample() # Create an object
    X.set_value(1, 2)
    # print(mean(X))
    print("Mean Value: ", mean(X))